### PR TITLE
pylint Makefile tests

### PR DESCRIPTION
**Pull request by TomConlin**
Firsts steps towards  shared static checking by default.

not enabled/enforced to begin with (tests would fail)

At minimum I would like to see `make lint_error` always pass
and move towards `make lint_warn` producing minimal output.

 - `.pylintrc` may be edited to tune for our collective tolerance for cruft
 - specialized Malkfile rules can be added to suit particular cases.

### CODE REVIEW

**Comment by justaddcoffee** [comment]
@TomConlin made a slight change to `./scripts/within_pip_env.py`

Very much like/appreciate a set of pylint checks we can do in `Makefile` format

File: dipper/curie_map.yaml [hunk 1]
```diff
@@ -193,6 +193,7 @@
 'dictyBase': 'http://dictybase.org/gene/'                                       # Dictyostelium discoideum database (social amoebae)
 'EcoGene': 'http://ecogene.org/gene/'                                           # Escherichia coli K-1
 'ENSEMBL': 'http://ensembl.org/id/'                                             # joint EMBL, EBI, Sanger Institute central resource
+'EnsemblGenome': 'http://www.ensemblgenomes.org/id/'                            # The Ensembl genome annotation system
```
Comment (justaddcoffee) [inline dipper/curie_map.yaml]: Why the new curie prefix? (just curious)

**Comment by TomConlin** [comment]
It is disturbing that something so purposefully basic fails on a platform being used for development.

File: dipper/curie_map.yaml (hunk 1 above)
Comment (TomConlin) [reply dipper/curie_map.yaml]: they were there before I found them, just written out as literals.

**Pull request merged**